examples/static-site-gen/platform/src/lib.rs

```diff
@@ -300,3 +300,97 @@ fn is_roc_code_block(cbk: &pulldown_cmark::CodeBlockKind) -> bool {
         }
     }
 }
+
+fn push_html_span(mut buf: Vec<String>, curr: &str, class: &str) -> Vec<String> {
+    // html escape strings from source code
+    let escaped = html_escape::encode_text(curr);
+
+    buf.push(format!("<span class=\"{}\">{}</span>", class, escaped));
+
+    buf
+}
+
+fn push_html(mut buf: Vec<String>, curr: &str) -> Vec<String> {
+    // html escape strings from source code
+    let escaped = html_escape::encode_text(curr);
+
+    buf.push(format!("{}", escaped));
+
+    buf
+}
+
+fn highlight_roc_code(code: &str) -> String {
+
+    use roc_region::all::{Loc, Region};
+    use roc_parse::highlight::{Token};
+
+    let locations: Vec<Loc<Token>> = roc_parse::highlight::highlight(code);
+    let mut buf: Vec<String> = Vec::new();
+    let mut offset = 0;
+
+    for location in locations {
+        let current_text = &code[offset..location.byte_range().end];
+
+        match location.value {
+            Token::LineComment | Token::DocComment => {
+                buf = push_html_span(buf, current_text, "comment");
+            }
+            Token::SingleQuote
+            | Token::String
+            | Token::UnicodeEscape
+            | Token::EscapedChar
+            | Token::Interpolated => {
+                buf = push_html_span(buf, current_text, "str");
+            }
+            Token::Keyword => {
+                buf = push_html_span(buf, current_text, "kw");
+            }
+            Token::Number => {
+                buf = push_html_span(buf, current_text, "number");
+            }
+            Token::Pipe => {
+                buf = push_html_span(buf, current_text, "pipe");
+            }
+            Token::Arrow => {
+                buf = push_html_span(buf, current_text, "arrow");
+            }
+            Token::Backpass => {
+                buf = push_html_span(buf, current_text, "arrow");
+            }
+            Token::Comma => {
+                buf = push_html_span(buf, current_text, "comma");
+            }
+            Token::Colon | Token::Backslash => {
+                buf = push_html_span(buf, current_text, "colon");
+            }
+            Token::GreaterThan | Token::Minus | Token::LessThan | Token::Plus | Token::Equals => {
+                buf = push_html_span(buf, current_text, "op");
+            }
+            Token::Brace | Token::Bracket | Token::Paren => {
+                buf = push_html_span(buf, current_text, "paren");
+            }
+            _ => {
+                buf = push_html(buf, current_text);
+            }
+        }
+
+        offset = location.byte_range().end;
+    }
+
+    format!("<pre><samp>{}</samp></pre>", buf.join(""))
+}
+
+// CAN WE RUN Rust tests in a platform?
```

Comment:
Unsure about this. I'll post a question in Zulip. Basically, how do we do testing in a platform.